docs/getting-started/roles/sqlalchemy/basic_roles.rst

```diff
@@ -0,0 +1,270 @@
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/__init__.py
+    :class: copybutton
+    :language: python
+    :lines: 11-14, 57-66
+    :caption: :fab:`python` __init__.py
+
+
+
+
+Create a users model
+---------------------
+
+Add a user model that will represent your app's users (if you don't already have one).
+
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/models.py
+    :class: copybutton
+    :language: python
+    :lines: 9-10, 14-21
+    :caption: :fab:`python` models.py
+
+
+Create an organizations model
+------------------------------
+
+Add an organization model that will represent the organizations or tenants
+that users belong to. The roles you create will be scoped to this model. If
+your app isn't multi-tenant, you can create an ``Application`` model instead,
+with one instance used globally.
+
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/models.py
+    :class: copybutton
+    :language: python
+    :lines: 24-32
+    :caption: :fab:`python` models.py
+
+
+
+Add an endpoint that needs authorization
+----------------------------------------
+
+Create or choose an existing endpoint that will need authorization to access.
+In our sample app we've created 2 endpoints that have different authorization
+requirements: one to view repositories and another to view billing
+information.
+
+Add policy checks to your code to control access to the protected endpoints.
+
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/routes.py
+    :class: copybutton
+    :language: python
+    :lines: 1-2, 5-6, 13-27
+    :caption: :fab:`python` routes.py
+
+Our example uses :py:func:`flask_oso.FlaskOso.authorize` to complete the
+policy check. If you're not using Flask, check out our general-purpose
+:doc:`Python package </using/libraries/python/index>`.
+
+2. Add roles
+============
+
+Create the OrganizationRole class using the role mixin
+------------------------------------------------------
+
+The oso SQLAlchemy library provides a method to generate a mixin which
+creates a role model. Create the mixin by passing in the base, user, and
+organization models, as well as the role names. Then create a role model that
+extends it.
+
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/models.py
+    :class: copybutton
+    :language: python
+    :lines: 6, 50-58
+    :caption: :fab:`python` models.py
+
+
+
+Specify role permissions
+------------------------
+
+To give the roles permissions, write an oso policy.
+
+Since we already called :py:func:`sqlalchemy_oso.roles.enable_roles` in our ``init_oso()`` method,
+you can write Polar ``role_allow`` rules over ``OrganizationRoles``.
+
+.. literalinclude:: /examples/roles/sqlalchemy/oso-sqlalchemy-roles-guide/app/authorization.polar
+    :class: copybutton
+    :language: polar
+    :lines: 7-14
+    :caption: :fa:`oso` authorization.polar
```

Comment:
Phrase the comments in here consistently.

"The member role can..."

"The billing role can..."